### PR TITLE
ci: add kind node image matrix for end to end testing

### PR DESCRIPTION
**Pull request by L3o-pold**
## This PR

- Add kind node image matrix for end to end testing

### Related Issues

Following the discussion here: https://github.com/open-feature/open-feature-operator/issues/500

### CODE REVIEW

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -119,6 +119,9 @@ jobs:
   e2e-test:
     runs-on: ubuntu-latest
     needs: docker-local
+    strategy:
+      matrix:
+        node_image: [ v1.21, v1.22, v1.23, v1.24, v1.25, v1.26, v1.27 ]
```
Comment (toddbaert) [inline .github/workflows/pr-checks.yml:124]: I have concerns with the breadth of this matrix, both due to resource consumption and supportability. 1.24 support ends in 3 days, I suggest we just do the oldest/newest supported, so reduce the matrix to `[v1.25, v1.27]`

Comment (L3o-pold) [reply .github/workflows/pr-checks.yml:124]: the goal is to give it a try to so which version is currently supported and fixeable.

File: go.mod [hunk 2]
```diff
@@ -10,24 +10,24 @@ require (
...
-	k8s.io/api v0.26.3
-	k8s.io/apiextensions-apiserver v0.26.1
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v0.26.3
-	sigs.k8s.io/controller-runtime v0.14.6
+	k8s.io/api v0.27.2
+	k8s.io/apiextensions-apiserver v0.27.2
+	k8s.io/apimachinery v0.27.2
+	k8s.io/client-go v0.27.2
+	sigs.k8s.io/controller-runtime v0.15.0
```
Comment (toddbaert) [inline go.mod:17]: Can we revert the dependency changes?

**Review by mowies — APPROVED** [review]
LGTM!

**Pull request merged**